app/src/main/java/org/mozilla/focus/browser/BrowserNavigationOverlay.kt

```diff
@@ -266,18 +266,30 @@ class BrowserNavigationOverlay @JvmOverloads constructor(
             else -> R.id.navButtonForward
         }
 
+        var viewToFocus = getLeftmostActiveItemInTopNav()
+        navUrlInput.nextFocusUpId = viewToFocus.id
+
         // We may have lost focus when disabling the focused view above.
         val isFocusLost = focusedView != null && findFocus() == null
         if (isFocusLost) {
             navUrlInput.requestFocus()
+            viewToFocus.requestFocus()
         }
 
         // Update nextFocusUpId of URL bar to leftmost active item in topNavContainer
         topNavContainer?.getFocusables(FOCUS_LEFT)?.get(0)?.let {
             navUrlInput.nextFocusUpId = it.id
+            maybeUpdateOverlayURLForCurrentState()
```

Comment:
I think this is misplaced in the wrong bracket - in this case, we're calling maybeUpdate only when we're successfully setting a focus, which shouldn't be a dependency.

Reply:
The whole block is unnecessary actually, it's already handled above. Github rebase issue perhaps? Removed!